imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/IMAPlugin.java

```diff
@@ -775,6 +806,52 @@ public void contentCompleted() {
         }
     }
 
+    @Override
+    public void playAdNow(String adTagUrl) {
+        if (isAdvertisingConfigured) {
+            isAdvertisingConfigLoading = !TextUtils.isEmpty(adTagUrl);
+            requestAdsFromIMA(null, adTagUrl);
+        }
+    }
+
+    @Override
+    public void setAdvertisingConfig(boolean isConfigured, IMAEventsListener imaEventsListener) {
+        isAdvertisingConfigured = isConfigured;
+        this.imaEventsListener = imaEventsListener;
+    }
+
+    @Override
+    public void setCuePoints(List<Long> cuePoints, AdBreakPositionType adBreakPositionType, boolean isUpdatedCuePoint) {
+        if (isAdvertisingConfigured && cuePoints != null && !cuePoints.isEmpty()) {
+            advertisingConfigCuePoints = cuePoints;
```

Comment:
@GouravSna do you clean it up for change media?

Reply:
Done,